### ISSUE TITLE
Issues on scene name when creating a SceneLoadData/LookupData with Scene reference.

### ISSUE THREAD

**Issue by FirstGearGames** [FirstGearGames/FishNet#437]
When creating a SceneLoadData or SceneLookupData using a scene reference which has a '.' in the name anything after the period is dropped off.

~~This is believed to be a Unity bug but will be investigated.~~
Not a Unity bug. Issue found and will be resolved next update.

**Comment by AlmostInteractive** [comment]
Just came here to report it.  It looks like replacing GetFileNameWithoutExtension with ChangeExtension is the way to go.  Or something similar.  Then the key is the whole path, not just the non-unique file name.

**Comment by FirstGearGames** [comment]
> ChangeExtension

Makes sense. Thanks for the feedback.

**Comment by FirstGearGames** [comment]
I'm going to actually use this...
```cs
                string modified = item;
                string extension = ".unity";
                int extIndex = modified.ToLower().IndexOf(extension);
                if (extIndex != -1 && (item.Length - extIndex) == extension.Length)
                    modified = modified.Substring(0, extIndex);

                result.Add(CreateData(modified));
```
There could be some cases where the extension simply does not exist if passing in by name. Using ChangeExtension on the example I gave such as sceneName.Vol2 with no '.scene' extension would remove the wrong content.

**Comment by FirstGearGames** [comment]
Resolved in 3.10.6
